### PR TITLE
Transformations: Add an All Unique Values Reducer

### PR DESCRIPTION
**Pull request by josiahg**
In the Group by transformation (and all other transformations that use reducers), there is an "All Values" calculation, and it would be useful to add a "All unique values" calculation. So if "All values" returns this array:

[1, 2, 3, 1, 2]

"All Unique Values" would return only:

[1, 2, 3]

### CODE REVIEW

File: packages/grafana-data/src/transformations/fieldReducer.ts [hunk 1]
```diff
@@ -237,6 +238,15 @@ export const fieldReducers = new Registry<FieldReducerInfo>(() => [
     standard: false,
     reduce: (field: Field) => ({ allValues: field.values.toArray() }),
   },
+  {
+    id: ReducerID.allUniqueValues,
+    name: 'All unique values',
+    description: 'Returns an array with all unique values',
+    standard: false,
+    reduce: (field: Field) => ({
+      allUniqueValues: field.values.toArray().filter((n, i) => field.values.toArray().indexOf(n) === i),
```
Comment (ryantxu) [inline packages/grafana-data/src/transformations/fieldReducer.ts:247]:
```suggestion
      allUniqueValues: [...new Set(field.values.toArray())],
```
@leeoniya -- can you resolve the best way to do this? :)

File: packages/grafana-data/src/transformations/fieldReducer.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export enum ReducerID {
   allIsZero = 'allIsZero',
   allIsNull = 'allIsNull',
   allValues = 'allValues',
+  allUniqueValues = 'allUniqueValues',
```
Comment (ryantxu) [inline packages/grafana-data/src/transformations/fieldReducer.ts:28]:
```suggestion
  uniqueValues = 'uniqueValues',
```

I don't think we need the "all" prefix -- it is used with the other boolean checks

**Comment by ryantxu** [comment]
@josiahg -- I love it, but I don't want to scare all the product managers into dusting off their typescript skills just to get stuff done 😜 

I added a simple test, and will let @leeoniya pick the best approach

Can you add a milestone?

**Comment by josiahg** [comment]
@ryantxu based on how my first attempt at creating this PR went, I don't think you have to worry about PMs doing too many of these 😂 

Your new suggested approach is elegant. I originally tried `new Set(field.values.toArray())` but found it still needed to be an array, went back to the drawing board (ok StackOverflow) and ended up with my approach. If @leeoniya approves, I think yours is much more idiomatic and easier to see what's going on.

File: packages/grafana-data/src/transformations/fieldReducer.ts (hunk 1 above)
Comment (leeoniya) [reply packages/grafana-data/src/transformations/fieldReducer.ts:247]: 👍 for the Set / spread strategy

**Review by leeoniya — APPROVED** [review]

**Pull request merged**